pkg/client/unversioned/testclient/fixture.go

```diff
@@ -54,13 +54,22 @@ type ObjectScheme interface {
 // Object or error that matches the requested action. For instance, list-replicationControllers
 // should attempt to return a list of replication controllers. This method delegates to the
 // ObjectRetriever interface to satisfy retrieval of lists or retrieval of single items.
-// TODO: add support for sub resources
 func ObjectReaction(o ObjectRetriever, mapper meta.RESTMapper) ReactionFunc {
 
 	return func(action Action) (bool, runtime.Object, error) {
-		kind, err := mapper.KindFor(unversioned.GroupVersionResource{Resource: action.GetResource()})
-		if err != nil {
-			return false, nil, fmt.Errorf("unrecognized action %s: %v", action.GetResource(), err)
+		var err error
+		var kind unversioned.GroupVersionKind
+
+		if action.GetSubresource() != "" {
+			kind, err = mapper.KindFor(unversioned.GroupVersionResource{Resource: action.GetSubresource()})
```

Comment:
> Subresource and resource shouldn't be treated as identical like this - they're not the same thing. @deads2k I don't think we ever decided what to do with sub resources here - do you remember?

They aren't same, but we never sorted it out since the CLI (main restmapper consumer) didn't speak to them.